test/rules/prefer_int_value_is_even_check.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_int_value_is_even_check`
+
+bool isEven = 1 % 2 == 0; //LINT
+bool isOdd = 13 % 2 == 1; //LINT
+
+bool a = 1 % 2 >= 0; //LINT
```

Comment:
This example is concerning. The modulo operator will always return a non-negative value, so the condition `>= 0` is always `true`. Replacing this with an invocation of either `isEven` or `isOdd` would not preserve semantics, so we shouldn't produce a diagnostic recommending that change.

We either need to add logic to the lint rule to ensure that the diagnostic is only produced when the inequality can validly be replaced by one of those invocations, or we need to change the lint rule so that it only uses the equality operators.

But I'll note that even with the equality operators we can have the same situation. Consider `a % 2 == 3`, which is always `false` and similarly shouldn't produce a diagnostic.

Reply:
sound good to me. yes, so i simplify this lint check to only warn %2 ==0 and %2 ==1. other cases would not yield a diagnostic.